tests/models/test_requests.py

```diff
@@ -73,6 +73,18 @@ async def streaming_body(data):
     assert request.headers["Transfer-Encoding"] == "chunked"
 
 
+def test_ignore_transfer_encoding_header():
```

Comment:
Let's go for `test_ignore_transfer_encoding_header_if_content_length_exists`, and include a docstring linking to the issue that this resolves.